ecl/hthor/hthor.cpp

```diff
@@ -457,22 +469,32 @@ void CHThorDiskWriteActivity::stop()
         uncompressedBytesWritten = outSeq->getPosition();
     close();
     updateWorkUnitResult(numRecords);
-    if((helper.getFlags() & (TDXtemporary | TDXjobtemp) ) == 0 && !agent.queryResolveFilesLocally())
+    if((helperFlags & (TDXtemporary | TDXjobtemp) ) == 0 && !agent.queryResolveFilesLocally())
         publish();
     incomplete = false;
     if(clusterHandler)
         clusterHandler->finish(file);
     CHThorActivityBase::stop();
-    if (helper.getFlags() & TDXvarfilename)
+    if (helperFlags & TDXvarfilename)
         filename.clear();
 }
 
 void CHThorDiskWriteActivity::resolve()
 {
     OwnedRoxieString rawname = helper.getFileName();
-    mangleHelperFileName(mangledHelperFileName, rawname, agent.queryWuid(), helper.getFlags());
+    mangleHelperFileName(mangledHelperFileName, rawname, agent.queryWuid(), helperFlags);
     assertex(mangledHelperFileName.str());
-    if((helper.getFlags() & (TDXtemporary | TDXjobtemp)) == 0)
+
+    if (useGenericReadWrites)
+    {
+        // Merge options from helper into providerOptions and formatOptions
+        CPropertyTreeWriter providerOptsWriter(providerOptions);
```

Comment:
Minor: If this could be used in a child query (not sure that ever happens), then this would need to start from a fresh tree.  (Also you can test TDXdynprovideroptions, TDXdynformatoptions and avoid recreating if they are not set in that case.)

Reply:
Now copies both options only if their respective enums are set.

Not sure what you mean by "start from a fresh tree".